manuscripts2/Readme.md

```diff
@@ -0,0 +1,230 @@
+# Readme
+
+This folder is an initialization to the new functions and classes that have to be added to manuscripts. This file provides a brief introduction to these classes and functions.
+
+
+### TODO:
+- Is popping from dict and then creating the nested aggregation the most efficient way to do this?
+- How do you parse 3 level nested aggregation? 
+
+
+### Structure:
+The main Class which is being implemented is the `Query` class. `Query` provides a connection to elasticsearch, queries it and computes the response.
+
+The important variables inside the `Query` objects are as follows:
+
+- `search`: This is the elasticsearch_dsl Search object which is responsible of quering elasticsearch and getting the results. All the aggregations, queries and filters are applied to this Search object and then it queries elasticsearch and fetches the required results.
+- `parent_agg_counter`: This is a counter counting the aggregations that are applied to the Search object. It starts with `0` and increments as aggregations are added.
+- `queries`: This is a dictionary containing two lists: `must` and `must_not`. This dictionary contains the normal(`must`) and inverse(`must_not`) queries.
+- `aggregations`: This is an OrderedDict which contains all the aggregations applied. An ordered dict allows us to create nested aggregations easily, as we'll see below.
+- `child_agg_counter_dict`: This dict keeps a track of the number of child aggregations that have been applied in an aggregation.
+
+Rest of the variables are self explainatory.
+
+`PullRequests` and `Issues` are subclasses derieved from the `Query` object. They have the initial queries: `"pull_requests":"true"` and `"pull_requests":"false"` respectively. They will have class specific functions in the future as the definitions of the metrics becomes clear.
+
+
+### Usage
+
+##### EXAMPLE 1: Basic usage
+
+The idea is that the user can use the chainability of functions to create nested aggs and add queries seamlessly.
+
+```python
+from new_functions import Query, Index
```

Comment:
See my notes on the Query class. My idea for this is more like this would be used this way:

```
sample = Query(github_index) \
    .add_query({"name1":"value1"}) \
    .add_inverse_query({"name2":"value2"})
```